bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandler.java

```diff
@@ -183,4 +186,13 @@
      * Only then it will be removed completely.
      */
     void handleRemoval();
+
+    /**
+     * This method provides a list of classes which should be registered as services by the framework
+     *
+     * @return - list of classes that will be registered as OSGi services
+     */
+    default Collection<Class> getServices() {
```

Comment:
you may type `Class` by `<? extends ThingHandlerService>`. This will remove the warning and also make sure the returned list are classes which implement THS.